### PR TITLE
Avoid crash on gui when rendering from command line and no write access

### LINKED ISSUE #2902: trying to show a QMessageBox when no gui is present in command line
when using command line for rendering if the output or config file aren't writable, it can't create the QWidget showing the error message, because there is no gui
What do you think of including `GuiApplication.h` on the respective classes, and `if (gui)` shows the QMessageBox, or else a console message?

### PR DESCRIPTION
**Pull request by zonkmachine**
Fixes #2902

Something like this?

### CODE REVIEW

**Comment by zonkmachine** [comment]
The other option is to fix it in [AudioFileDevice.cpp](https://github.com/LMMS/lmms/blob/master/src/core/audio/AudioFileDevice.cpp#L51:L64)
Add "GuiApplication.h" as suggested in #2902 printf something and EXIT_FAILURE when no gui. Both works just fine.

**Comment by zonkmachine** [comment]
Pushed method two on a separate commit for comparison. I vote no. 2

File: src/core/audio/AudioFileDevice.cpp [hunk 1]
```diff
@@ -50,17 +51,26 @@ AudioFileDevice::AudioFileDevice( const sample_rate_t _sample_rate,
...
 	{
-		QMessageBox::critical( NULL,
-			ExportProjectDialog::tr( "Could not open file" ),
-			ExportProjectDialog::tr( "Could not open file %1 "
+		QString title, message;
+		title = ExportProjectDialog::tr( "Could not open file" );
+		message = ExportProjectDialog::tr( "Could not open file %1 "
 						"for writing.\nPlease make "
 						"sure you have write-"
 						"permission to the file and "
```
Comment (zonkmachine) [inline src/core/audio/AudioFileDevice.cpp]: `write-permission`?

-> `write permission`

**Comment by zonkmachine** [comment]
Settled for no.2 . Done... :steam_locomotive:

File: src/core/audio/AudioFileDevice.cpp [hunk 2]
```diff
@@ -50,17 +51,27 @@ AudioFileDevice::AudioFileDevice( const sample_rate_t _sample_rate,
...
+
+		if( gui )
+		{
+			QMessageBox::critical( NULL, title, message,
+						QMessageBox::Ok,
+						QMessageBox::NoButton );
+		}
+		else
+		{
+			printf( "%s\n", message.toUtf8().constData() );
```
Comment (jasp00) [inline src/core/audio/AudioFileDevice.cpp]: `fprintf( stderr, "%s\n",`

Comment (zonkmachine) [inline src/core/audio/AudioFileDevice.cpp]: Fixed!

**Comment by zonkmachine** [comment]
Thanks! Merge...

**Pull request merged**